stl/inc/xlocnum

```diff
@@ -251,6 +251,230 @@ __PURE_APPDOMAIN_GLOBAL locale::id numpunct<_Elem>::id;
 #endif // __clang__
 #endif // !defined(_CRTBLD) || defined(CRTDLL2) || !defined(_DLL) || defined(_M_CEE_PURE)
 
+// TRANSITION, ABI: Sentinel value used by num_get::do_get() to enable correct "V2" behavior
+#define _ENABLE_V2_BEHAVIOR 1000000000
+
+template <class _Elem, class _InIt>
+int _Parse_fp_with_ios_base(char* const _Ac, _InIt& _First, _InIt& _Last, ios_base& _Iosbase, int* _Phexexp) {
+    // get floating-point field from [_First, _Last) into _Ac
+    const auto& _Punct_fac = _STD use_facet<numpunct<_Elem>>(_Iosbase.getloc());
+    char* _Ptr             = _Ac;
+
+    constexpr size_t _Offset_dec_digit_end = 10;
+    constexpr size_t _Offset_hex_digit_end = 22;
+    constexpr size_t _Offset_neg_sign      = 22;
+    constexpr size_t _Offset_pos_sign      = _Offset_neg_sign + 1;
+    constexpr size_t _Offset_upper_x       = 24;
+    constexpr size_t _Offset_lower_x       = _Offset_upper_x + 1;
+    constexpr size_t _Offset_upper_p       = 26;
+    constexpr size_t _Offset_lower_p       = _Offset_upper_p + 1;
+    constexpr size_t _Offset_upper_e       = 0xE;
+    constexpr size_t _Offset_lower_e       = 0xE + 0x10 - 10;
+    static constexpr char _Src[]           = "0123456789ABCDEFabcdef-+XxPp";
+    _Elem _Atoms[sizeof(_Src)];
+    const ctype<_Elem>& _Ctype_fac = _STD use_facet<ctype<_Elem>>(_Iosbase.getloc());
+    _Ctype_fac.widen(_STD begin(_Src), _STD end(_Src), _Atoms);
+
+    const _Elem _Positive_sign = _Atoms[_Offset_pos_sign];
+    const _Elem _Negative_sign = _Atoms[_Offset_neg_sign];
+    const _Elem _Zero_wc       = _Atoms[0];
+
+    if (_First != _Last) {
+        if (*_First == _Positive_sign) { // gather plus sign
+            *_Ptr++ = '+';
+            ++_First;
+        } else if (*_First == _Negative_sign) { // gather minus sign
+            *_Ptr++ = '-';
+            ++_First;
+        }
+    }
+
+    *_Ptr++ = '0'; // backstop carries from sticky bit
+
+    bool _Parse_hex                = false;
+    char _Initial_dec_leading_zero = '\0';
+    if (_First != _Last && *_First == _Zero_wc) {
+        ++_First;
+        if (_First == _Last) { // "0" only
+            *_Ptr = '\0';
+            return 0;
+        }
+        if (*_First == _Atoms[_Offset_lower_x] || *_First == _Atoms[_Offset_upper_x]) { // 0X or 0x
+            _Parse_hex = true;
+            ++_First; // discard 0X or 0x for further parsing
+            *_Ptr++ = 'x';
+        } else {
+            ++_Initial_dec_leading_zero;

```

Comment:
If one tries to parse 128 leading zeroes, this will fail; seems like we should check for this case.

Reply:
`_Initial_dec_leading_zero` is incresed at most once, and it is only used to form the initial value of `_Groups` (of type `string`). The remain leading zeroes is counted later.